junit5/src/main/java/cz/xtf/junit5/extensions/OpenShiftRecorderHandler.java

```diff
@@ -1,220 +1,54 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import cz.xtf.core.bm.BuildManagers;
-import cz.xtf.core.openshift.OpenShift;
-import cz.xtf.core.openshift.OpenShifts;
-import cz.xtf.junit5.annotations.OpenShiftRecorder;
 import cz.xtf.junit5.config.JUnitConfig;
-import cz.xtf.junit5.extensions.helpers.EventsFilterBuilder;
-import cz.xtf.junit5.extensions.helpers.ResourcesFilterBuilder;
-import cz.xtf.junit5.extensions.helpers.ResourcesPrinterHelper;
-import cz.xtf.junit5.extensions.helpers.ResourcesTimestampHelper;
-import io.fabric8.kubernetes.api.model.ConfigMap;
-import io.fabric8.kubernetes.api.model.Event;
-import io.fabric8.kubernetes.api.model.HasMetadata;
-import io.fabric8.kubernetes.api.model.Pod;
-import io.fabric8.kubernetes.api.model.Secret;
-import io.fabric8.kubernetes.api.model.Service;
-import io.fabric8.kubernetes.api.model.apps.StatefulSet;
-import io.fabric8.openshift.api.model.Build;
-import io.fabric8.openshift.api.model.BuildConfig;
-import io.fabric8.openshift.api.model.DeploymentConfig;
-import io.fabric8.openshift.api.model.ImageStream;
-import io.fabric8.openshift.api.model.Route;
 
 /**
- * Record OpenShift isolated state relative to a test.
- * Resources are filtered by name provided via {@link OpenShiftRecorder} annotation. Names are turned into regexes
- * by adding {@code .*} as a suffix. If no name is provided, resources in namespaces (BM and master) are filtered automatically
- * by recording what resources are seen before test and so on.
- * <p>
- * Recorded resources:
- * <ul>
- * <li>pods states</li>
- * <li>deployment configs states</li>
- * <li>builds states</li>
- * <li>build configs states</li>
- * <li>image streams states</li>
- * <li>stateful sets states</li>
- * <li>routes states</li>
- * <li>services states</li>
- * <li>secrets states</li>
- * <li>logs of pods</li>
- * <li>events</li>
- * </ul>
- * <p>
- * OpenShift state is recorded when a test throws an exception. If {@link JUnitConfig#recordAlways()} is true, state is
- * recorded also when a test passes.
- * <p>
- * Use {@link JUnitConfig#recordDir()} ()} to set the directory of records.
+ * Implements JUnit lifecycle events handling in order to record OCP state for different purposes (e.g.: debug).
+ * The operational logic to perform the steps to get the status from OCP is delegated to
+ * {@link OpenShiftRecorderService}.
+ *
+ * Based on the type of the event that is being handled, an instance of this class will either call
+ * {@link OpenShiftRecorderService#initFilters(ExtensionContext)} or
+ * {@link OpenShiftRecorderService#resetFilters(ExtensionContext)} (e.g. on
+ * {@link BeforeAllCallback#beforeAll(ExtensionContext)} or on {@link BeforeEachCallback#beforeEach(ExtensionContext)})
+ * or instead request to get an OCP state record via {@link OpenShiftRecorderService#recordState(ExtensionContext)}, eg.
+ * when handling {@link TestExecutionExceptionHandler#handleTestExecutionException(ExtensionContext, Throwable)}
```

Comment:
Can you point out that the recording is bound to:
- `handleTestExecutionException`
- `handleBeforeAllMethodExecutionException` / `handleBeforeEachMethodExecutionException` - only if  `xtf.record.before` is set
- `testSuccessful` only if `xtf.record.always` is set
?

Reply:
Sure, I will, thanks.

Reply:
Done.